fluidkeys/main.go

```diff
@@ -267,6 +270,63 @@ func keyCreate() exitCode {
 	return 0
 }
 
+func keyList() exitCode {
+	fluidkeysDirectory, err := getFluidkeysDirectory()
+	if err != nil {
+		fmt.Printf("Failed to get fluidkeys directory")
+	}
+	db := database.New(fluidkeysDirectory)
+
+	keys, err := loadPgpKeys(db)
+	if err != nil {
+		panic(err)
+	}
+
+	emailAddressColumnWidth := len("Email address")
+
+	for _, key := range keys {
+		for _, id := range key.Identities {
+			emailAddressColumnWidth = Max(emailAddressColumnWidth, len(id.Name))
+		}
+	}
+
+	createdColumnWidth := 11
+	nextRotationColumnWidth := 13
+
+	header := fmt.Sprintf("%-*s  ", emailAddressColumnWidth, "Email address")
+	header += fmt.Sprintf("%-*s  ", createdColumnWidth, "Created")
+	header += fmt.Sprintf("%-*s  ", nextRotationColumnWidth, "Next rotation")
+
+	fmt.Printf("%s\n", colour.LightBlue(header))
+
+	printHorizontalRule(emailAddressColumnWidth, createdColumnWidth, nextRotationColumnWidth)
+
+	for _, key := range keys {
+		firstRow := true
+		for id := range key.Identities {
+			if firstRow {
+				fmt.Printf("%-*s  ", emailAddressColumnWidth, id)
+				fmt.Printf("%-*s  ", createdColumnWidth, key.PrimaryKey.CreationTime.Format("2 Jan 2006"))
+				fmt.Printf("\n")
+				firstRow = false
+			} else {
+				fmt.Printf("%s\n", id)
+			}
+
+		}
+		printHorizontalRule(emailAddressColumnWidth, createdColumnWidth, nextRotationColumnWidth)
+	}
+
+	return 0
+}
+
+func printHorizontalRule(columnWidths ...int) {
```

Comment:
I wonder if the name is *slightly* misleading, because it's not a "horizontal rule" as I imagine from HTML, it's more a series of underlines, maybe it should be called `printHorizontalUnderlines` ?